gemrb/core/Projectile.cpp

```diff
@@ -250,6 +250,8 @@ void Projectile::Setup()
 		ZPos = ProHeights::Flying;
 	}
 
+	ZPos /= 2;
+
```

Comment:
What projectile does this address? And the ZPos halving is mighty suspicious.

Reply:
> What projectile does this address? And the ZPos halving is mighty suspicious.

All arrows/bullets. They flew from the character's head. Division by two is visually correct.

Reply:
It should be fixed in the attack projectile then, when it's created, not here where it affects also spells.

Reply:
I highly doubt it. The casting offsets for the casting glow positions were hardcoded and the community managed to recover that fully, which we now implement. The spells are launched from the same points — if some aren't that's a bug that needs to be addressed. Hence the perpetual need for examples, which also help when code gets revisited.

Reply:
Done refix it

Reply:
Testing a bit, it might have actually used the same position as spells. Try using Projectile::GetStartOffset to adjust Pos (no need to touch ZPos manually then) and see if it looks ok.

Reply:
Ok

Reply:
start.y += ZPos/2 ? GetStartOffset static function, i cant read ZPos. Also, it's not very pretty code

Reply:
Something like `attackProjectile->Pos += Projectile::GetStartOffset(this);`.

Reply:
GetStartOffset returns negative values; if you add it, it raises the projectile even higher. You can subtract minus by minus, it seems true, but you still need to change the final position on the target, otherwise it flies much higher. Working with ZPos made editing easier.

Reply:
ok, let's try the halving route then.

Reply:
Projectile::GetStartOffset has almost no effect on the height of the attacking projectile (+1/-1).
But have effect pos.y-=GetZPos(); in DrawTravel, it does not separate caster projectiles and attacking ones.
I think I need to work with ZPos in this situation.

Reply:
1) GetStartOffset has almost no effect on projectile height (+1/-1)
2) The height of the projectile is affected by pos.y-=GetZPos(); in the DrawTravel function
3) With ZPos 0 projectile height on center of the character circle.
4) With default ZPos projectile height on above the character's head.

I think the problem is in ZPos.

Reply:
Since there's no Z axis, ZPos is just an offset to Pos.y (your second point), that's why it was worth a shot. I was expecting Projectile::MoveTo to apply it already, because the caster should be set by then. So I don't understand what's different. Are you getting the normal or flying height assigned to ZPos? The latter is almost double the first, which could explain things ... does adding `attackProjectile->SFlags|= PSF_FLYING;` work too?

Reply:
Attack projectiles already have the PSF_FLYING flag set, it reads it from bullet.pro. Resetting this flag reduces ZPos from ProHeights::Flying to ProHeights::Normal, which looks good for normal sized characters, but is still bad for hobbits and dwarves.

Reply:
Aren't smaller/bigger creatures worse off anyway, since we use a constant? Yuan-ti archers are the biggest ranged unit that comes to mind (it'd be amazing if they used special ammo just to compensate).

Reply:
I looked at the original and there is the height of the arrow for hobbits and dwarves
bad too)

Reply:
We can fix this once we externalize all the animation data and can just add another key. Now only pst uses it. #93 
Ok, we could add another table in the interim if anyone cared that much.
For now then it's simpler if we just remove the PSF_FLYING flag, since it's a smaller change and you say it works as well. Thanks for the investigation!

Reply:
I do what I can)